sdk/eventhub/event-hubs/src/eventHubReceiver.ts

```diff
@@ -557,12 +586,63 @@ export class EventHubReceiver extends LinkEntity {
     );
   }
 
+  private async _drainAndAddCredits(
+    onMessage: OnMessage,
+    abortSignal?: AbortSignalLike
+  ): Promise<void> {
+    await delay(0);
+    this._isDrainingQueue = true;
+    while (this.internalQueue.length) {
+      if (!this._hasActiveListeners()) {
+        this._isDrainingQueue = false;
+        return;
+      }
+
+      if (abortSignal && abortSignal.aborted) {
+        this._isDrainingQueue = false;
+        return;
+      }
+
+      if (this._userDefinedOnMessage !== onMessage) {
+        this._isDrainingQueue = false;
+        return;
+      }
+      const eventData = this.internalQueue.splice(0, 1)[0];
+      onMessage(eventData);
+      // allow the event loop to process any blocking code outside
+      // this code path before sending the next event.
+      await delay(0);
+    }
+    this._isDrainingQueue = false;
+
+    if (this._hasActiveListeners()) {
+      // register the onMessage handler to stop
+      // sending events to the queue.
+
+      if (this.isOpen()) {
+        // add credits to start receiving events from the service.
+        this._addCredit(this.prefetchCount || 1);
+      }
+    }
+  }
+
+  private _existingCredits(): number {
+    if (this._receiver) {
+      return this._receiver.credit;
+    }
+    return 0;
+  }
```

Comment:
This function doesn't add much value. It is used only once. Inlining it would be better

Reply:
Removed!